src/Aspire.Hosting.Azure/Provisioning/Internal/DefaultProvisioningContextProvider.cs

```diff
@@ -26,8 +30,106 @@ internal sealed class DefaultProvisioningContextProvider(
 {
     private readonly AzureProvisionerOptions _options = options.Value;
 
+    private readonly TaskCompletionSource _provisioningOptionsAvailable = new();
+
+    private void EnsureProvisioningOptions(JsonObject userSecrets)
+    {
+        if (!string.IsNullOrEmpty(_options.Location) && !string.IsNullOrEmpty(_options.SubscriptionId))
+        {
+            // If both options are already set, we can skip the prompt
+            _provisioningOptionsAvailable.TrySetResult();
+            return;
+        }
+
+        if (interactionService.IsAvailable)
+        {
+            // Start the loop that will allow the user to specify the Azure provisioning options
+            _ = Task.Run(async () =>
+            {
+                try
+                {
+                    await RetrieveAzureProvisioningOptions(userSecrets).ConfigureAwait(false);
+
+                    logger.LogDebug("Azure provisioning options have been handled successfully.");
+                }
+                catch (Exception ex)
+                {
+                    logger.LogError(ex, "Failed to retrieve Azure provisioning options.");
+                }
+            });
+        }
+    }
+    private async Task RetrieveAzureProvisioningOptions(JsonObject userSecrets, CancellationToken cancellationToken = default)
+    {
+        while (_options.Location == null || _options.SubscriptionId == null)
+        {
+            var locations = typeof(AzureLocation).GetProperties(BindingFlags.Public | BindingFlags.Static)
+                            .Where(p => p.PropertyType == typeof(AzureLocation))
+                            .Select(p => (AzureLocation)p.GetValue(null)!)
+                            .Select(location => KeyValuePair.Create(location.Name, location.DisplayName ?? location.Name))
+                            .OrderBy(kvp => kvp.Value)
+                            .ToList();
+
+            var messageBarResult = await interactionService.PromptMessageBarAsync(
+                 "Azure Provisioning",
+                 "The model contains Azure resources that require an Azure Subscription.",
+                 new MessageBarInteractionOptions
+                 {
+                     Intent = MessageIntent.Warning,
+                     PrimaryButtonText = "Enter values"
+                 },
+                 cancellationToken)
+                 .ConfigureAwait(false);
+
+            if (messageBarResult.Canceled)
+            {
+                // User canceled the prompt, so we exit the loop
+                _provisioningOptionsAvailable.SetException(new MissingConfigurationException("Azure provisioning options were not provided."));
+                return;
+            }
+
+            if (messageBarResult.Data)
+            {
+                var result = await interactionService.PromptInputsAsync(
+                    "Azure Provisioning",
```

Comment:
In the dashboard we only capitalize the first word in a title.
```suggestion
                    "Azure provisioning",
```